src/cmd/flux-getrabbit.py

```
@@ -60,11 +66,22 @@ def main():
         sys.exit(f"File {path!r} could not be parsed as JSON: {jexc}")
     # construct and print the hostlist of rabbits
     hlist = Hostlist()
-    if not args.computes and not args.rabbits:
+    if not args.computes and not args.rabbits and not args.jobids:
         # print out all rabbits
         hlist.append(mapping["rabbits"].keys())
         print(hlist.uniq().encode())
         return
+    if args.jobids:
+        for jobid in args.jobids:
+            try:
+                JobID(jobid)
+            except Exception as exc:
+                sys.exit(f"Could not interpret {jobid} as a flux Jobid: {exc}")
+            try:
+                nodelist = flux.job.job_list_id(handle, jobid, ["nodelist"]).nodelist
```

I would assume you need to use `.get_jobinfo()` on the result of `job_list_id()` (a `JobListIdRPC`) to then get the `nodelist` attribute (but I may be missing something obvious).

Also, not critical, but you can get the nodelist for all jobids using one RPC via the `JobList` class. For example, something like:
```python
  jobs = JobList(handle, ids=jobids, attrs=["nodelist"]).jobs()
  for job in jobs:
    args.computes.append(job.nodelist)
```

---

No worries, it's my fault for force-pushing so much. And I probably should have kept the history better anyway. I'm going to set MWP though. The single RPC makes sense, but I imagine 99% of users will just pass one job ID, and it seemed like `JobList()` when I tried it just now failed silently on a bad job ID, whereas `job_list_id` raised an exception, and that may make for a better user experience.

---

Eh, I was wrong anyway, `JobList()` uses `job_list_id` under the covers when multiple, specific jobids are passed anyway.  (it just sends the RPCs all at once and returns a single future which is fulfilled when all RPCs are complete).

And agreed that the error handling is awkward (putting it politely) (errors are held in `self.errors`, which you have to check if less jobs were returned than expected)

Anyhow, sounds like you made the right choice here.